docs/guides/chip_tool_guide.md

```diff
@@ -0,0 +1,807 @@
+
+Before you can use the CHIP Tool, you must compile it from source on Linux
+(amd64/aarch64) or macOS. If you want to run it on Raspberry Pi, it must use a
+64-bit OS.
+
+> **Note:** To ensure compatibility, always build the CHIP Tool and the Matter
+> device from the same revision of the `connectedhomeip` repository.
+
+### Building the CHIP Tool
+
+To build and run the CHIP Tool:
+
+1. Install all required packages for Matter and prepare the source code and the
+   build system. Read the [Building Matter](BUILDING.md) guide for instructions.
+2. Open a command prompt in the `connectedhomeip` directory.
+3. Run the following command:
+
+    ```
+    ./scripts/examples/gn_build_example.sh examples/chip-tool BUILD_PATH
+    ```
+
+    In this command, `BUILD_PATH` specifies where the target binaries are to be
+    placed.
+
+### Running the CHIP Tool
+
+To check if the CHIP Tool runs correctly, execute the following command from the
+`BUILD_PATH` directory:
+
+```
+$ ./chip-tool
+```
+
+As a result, the CHIP Tool prints all available commands. These are called
+_clusters_ in this context, but not all listed commands correspond to the
+_clusters_ in the Data Model (for example, pairing or discover commands). Each
+listed command can however become the root of the new more complex command by
+appending it with sub-commands. Examples of specific commands and their use
+cases are described in the [Supported commands and options](#commands) section.
+
+<hr>
+
+<a name="using"></a>
+
+## Using CHIP Tool for Matter device testing
+
+This section describes how to use CHIP Tool to test the Matter device. The
+following steps depend on the application clusters that you implemented on the
+device.
+
+This tutorial is using the
+[Matter lighting app example](https://github.com/project-chip/connectedhomeip/tree/master/examples/lighting-app)
+with the Bluetooth LE commissioning method support. You can use other Matter
+examples and still follow this procedure. If you use a different example, the
+[Step 7](#step-7-control-application-data-model-clusters) may vary depending on
+the clusters implemented in your application.
+
+### Step 1: Prepare the Matter device
+
+Build and program the device with the Matter device firmware by following the
+[Matter lighting-app example](https://github.com/project-chip/connectedhomeip/tree/master/examples/lighting-app)
+documentation.
+
+### Step 2: Enable Bluetooth LE advertising on Matter device
+
+Some examples are configured to advertise automatically on boot. Other examples
+require physical trigger, for example pushing a button. Follow the documentation
+of the Matter device example to learn how Bluetooth LE advertising is enabled
+for the given example.
+
+### Step 3: Make sure the IP network is set up
+
+To follow the next steps, the IP network must be up and running. For instance,
+the Thread network can be established using
+[OpenThread Border Router](https://openthread.io/codelabs/openthread-border-router#0).
```

Comment:
```suggestion
To follow the next steps, the IP network must be up and running. This guide assumes that you are establishing the Thread network using
[OpenThread Border Router](https://openthread.io/codelabs/openthread-border-router#0).
```

Reply:
I would not change that because we are actually describing the WiFi use case later on.

Reply:
Yes, but in our conversation on Teams you said we're using Thread, and Wi-Fi is described as reference.

Reply:
Maybe I have not myself clear, sorry :) From our platform perspective the OpenThread is the use case, but others may use WiFi as well.